pkg/kubelet/kubelet_pods.go

```diff
@@ -1673,6 +1687,70 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 		statuses[container.Name] = status
 	}
 
+	for _, container := range containers {
```

Comment:
This block here is the immediate "fix" to the problem.  It indicates that when a pod is deleted, the container is not represented in status long enough to show an exit code, so there is no last termination status.